### DISCUSSION TITLE
oclif's eslint settings are pretty wild

### DISCUSSION THREAD

**Discussion by tmcw** [oclif/oclif#1406] [General]
I really like oclif, but my first run experience is simply that its `eslint` default is just incredibly prescriptive, stricter than any project I've encountered, and a lot of the rules are arguably aesthetic and naggy. Sorting pattern matching assignments alphabetically and adding spaces between local and global imports… regulating spaces between expressions, requiring JSDoc parameter definitions whenever you have a JSDoc comment. My impression is that it's a little over the top and suprising for a set of defaults.

**Comment by mdonnalley** [comment]
@tmcw Those eslint configs exist primarily for our repos and secondarily for the generated CLIs made by `oclif generate`

If you don't like the rules, you can always disable the ones you don't like or create your own eslint config with your preferred rules. There's nothing special about them that says you must use them in your oclif project.

If there's enough interest, we could create a separate, minimal eslint config that's used just for the templates. But until then, we'd prefer to not add another project that we have to maintain.
